roxie/ccd/ccdserver.cpp

```diff
@@ -7502,18 +7563,38 @@ class CRoxieServerHashDedupActivity : public CRoxieServerActivity
             return true;
         }
 
+        bool insertBest(const void * nextrow)
+        {
+            unsigned hash = helper.queryHash()->hash(nextrow);
+            const void *et = find(hash, nextrow);
+            if (et)
+            {
+                const HashDedupElement *element = reinterpret_cast<const HashDedupElement *>(et);
+                const void * row = element->queryRow();
+                if (bestCompare->docompare(row,nextrow) <= 0)
+                    return false;
+                removeExact( const_cast<void *>(et));
+                ReleaseRoxieRow(row);
+                // drop-through to add new row
+            }
+            addNew(new HashDedupElement(hash, nextrow), hash);
+            return true;
+        }
     private:
         IHThorHashDedupArg & helper;
         CachedOutputMetaData keySize;
         Owned<IEngineRowAllocator> keyRowAllocator;
         unsigned activityId;
+        ICompare * bestCompare;
     } table;
 
 public:
     CRoxieServerHashDedupActivity(IRoxieSlaveContext *_ctx, const IRoxieServerActivityFactory *_factory, IProbeManager *_probeManager)
-        : CRoxieServerActivity(_ctx, _factory, _probeManager), helper((IHThorHashDedupArg &)basehelper), table(helper, activityId)
+        : CRoxieServerActivity(_ctx, _factory, _probeManager), helper((IHThorHashDedupArg &)basehelper), table(helper, activityId), hashTableFilled(false), hashDedupTableIter(table)
     {
         eof = false;
+        keepBest = helper.getFlags() & DFkeepbest;
```

Comment:
Add a != 0 to avoid warnings on windows (and other?) compilers.  Also, extra trailing newline